packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -197,18 +206,19 @@ export const onEvent = async (event: AWSCDKAsyncCustomResource.OnEventRequest):
       result = {
         PhysicalResourceId: event.PhysicalResourceId,
       };
-      console.log('Fetching current table state');
-      const describeTableResultBeforeDeletion = await ddbClient.describeTable({ TableName: event.PhysicalResourceId });
-      if (describeTableResultBeforeDeletion.Table?.DeletionProtectionEnabled) {
-        // Skip the deletion when protection is enabled
-        return result;
-      }
       try {
+        console.log('Fetching current table state');
+        const describeTableResultBeforeDeletion = await ddbClient.describeTable({ TableName: event.PhysicalResourceId });
+        if (describeTableResultBeforeDeletion.Table?.DeletionProtectionEnabled) {
+          // Skip the deletion when protection is enabled
+          return result;
+        }
         console.log('Initiating table deletion');
         await ddbClient.deleteTable({ TableName: event.PhysicalResourceId });
         return result;
       } catch (err) {
-        if (err.code === 'ResourceNotFoundException') {
+        if (err instanceof ResourceNotFoundException) {
```

Comment:
This change is due to the AWS SDK v3 upgrade

Reply:
https://aws.amazon.com/blogs/developer/service-error-handling-modular-aws-sdk-js/